### DISCUSSION TITLE
Cant override category view

### DISCUSSION THREAD

**Discussion by Majega29** [templaza/astroid-framework#692] [General]
Hi,
i create a child template. So i would like to override the category view. I copy the libraries/astroid/framework/html in my template html and rename it and assign this layout to the category parameter layout.
It doesn't work. It is the default layout blog from libraries/astroid/framework/html.
Any help?
Please

**Comment by Chacapamac** [comment]
> — i create a child template.
> — i would like to override the category view
> — I copy the libraries/astroid/framework/html

First, I think that if you want to override a Content > category view, You have to look in the **Joomla Root > templates > YourTemplates > html > com_content > category>** if an override php file is already provided by your Astroid Template for whatever view you need to override.

• If You find one, create an **“html’** folder in your children first level Template and add a copy of that file to a **“com_content” > “category”** folder inside the “html” folder that you also create. 
Now you can override whatever **Category View** you want.

• If you don’t find any **Category View** given by your Astroid template you can copy the **Category View** you want to override from the **Joomla Root > components > com_content >  tmpl > category** folder to the same folder destination (see last point) and override from your chid template.

An idea of all **Category Views** available
![Screen Shot 2024-06-06 at 11 47 40 AM](https://github.com/templaza/astroid-framework/assets/1907276/db669f5a-4cef-4d43-bd21-b535423c3534)

**Reply by Majega29** [reply]
Ok thanks.
Is it a way to add a different layout for category view.
I want to create an specific override for sport teams (category) with joomla custom fields. 

So i duplicate the blog layout and call it "teams". I select it in the category parameters, but it is still the blog layout that is shown.

**Comment by Chacapamac** [comment]
What “Category parameters”? Here?

![Screen Shot 2024-06-07 at 8 32 13 AM](https://github.com/templaza/astroid-framework/assets/1907276/687ceab8-71b8-4200-9b16-999a5b943ab1)

**Reply by Chacapamac** [reply]
Yes, in Joomla, you can create your own custom layout for a category. Here's a high-level overview of the steps you would typically follow:

1. **Create a Template Override**:
   - Navigate to your template's directory and create an override for the `com_content` component's category view.
   - For example, if you're using the Protostar template, go to `templates/protostar/html/com_content/category/`.

2. **Duplicate an Existing Layout**:
   - Copy an existing layout file like `blog.php` and rename it to your custom layout name, such as `mycustomlayout.php`.

3. **Edit Your Custom Layout**:
   - Make the desired changes to your new layout file.

4. **Select Your Custom Layout in Joomla**:
   - In the backend, go to `Content` -> `Articles` -> `Options` -> `Category` and choose your custom layout from the 'Choose a layout' option.
   - Alternatively, when creating a new menu item for the category, you can select your custom layout under the 'Layout' option.

5. **Test Your New Layout**:
   - Create a menu item with your new category layout and see if it reflects the changes on the frontend.

Remember, if you want to have an alternate menu item type for your custom layout, you might need to review the section on Alternative Menu Items in Joomla's documentation on Layout Overrides¹.

For a more detailed guide, you can refer to Joomla's official documentation or community forums where you can find step-by-step instructions and troubleshooting tips¹². If you encounter any issues, these resources can be very helpful. Happy coding!

Source: Conversation with Copilot, 2024-06-07
(1) How do I create and use my own layout for category page? - Joomla .... https://forum.joomla.org/viewtopic.php?t=971175.
(2) How to create a custom category layout - Joomla! Forum - community .... https://forum.joomla.org/viewtopic.php?t=991281.
(3) Create Joomla Blog Layouts That Users Can Select - Joomlashack. https://www.joomlashack.com/blog/joomla/custom-category-layout/.
(4) getting-started-adding-a-category | Joomla! Documentation. https://docs-next.joomla.org/docs/getting-started/getting-started-adding-a-category/.
(5) Getting Started: Adding a Category - Joomla. https://docs.joomla.org/J4.x:Getting_Started:_Adding_a_Category.
(6) undefined. https://docs.joomla.org/How_to_override.
(7) undefined. https://docs.joomla.org/Layout_Overrides_in_Joomla.
(8) undefined. https://docs.joomla.org/Layout_Override.
(9) undefined. http://timb114.sg-host.com/home.
(10) undefined. https://ltheme.com/joomla-templates/.
(11) undefined. https://agethemes.com/free-joomla-templates/.
(12) undefined. https://forum.virtuemart.net/index.php?topic=130065.0.
